jupiterone/internal/client/generated.go

```diff
@@ -0,0 +1,1573 @@
+}
+
+// GetCreateQuestionRuleInstance returns CreateInlineQuestionRuleInstanceResponse.CreateQuestionRuleInstance, and is useful for accessing the field via an interface.
+func (v *CreateInlineQuestionRuleInstanceResponse) GetCreateQuestionRuleInstance() CreateInlineQuestionRuleInstanceCreateQuestionRuleInstance {
+	return v.CreateQuestionRuleInstance
+}
+
+// CreateQuestionCreateQuestion includes the requested fields of the GraphQL type Question.
+type CreateQuestionCreateQuestion struct {
+	Id              string                                                             `json:"id"`
+	Title           string                                                             `json:"title"`
+	Description     string                                                             `json:"description"`
+	PollingInterval SchedulerPollingInterval                                           `json:"pollingInterval"`
+	Queries         []CreateQuestionCreateQuestionQueriesQuestionQuery                 `json:"queries"`
+	Tags            []string                                                           `json:"tags"`
+	Compliance      []CreateQuestionCreateQuestionComplianceQuestionComplianceMetaData `json:"compliance"`
+}
+
+// GetId returns CreateQuestionCreateQuestion.Id, and is useful for accessing the field via an interface.
+func (v *CreateQuestionCreateQuestion) GetId() string { return v.Id }
+
+// GetTitle returns CreateQuestionCreateQuestion.Title, and is useful for accessing the field via an interface.
+func (v *CreateQuestionCreateQuestion) GetTitle() string { return v.Title }
+
+// GetDescription returns CreateQuestionCreateQuestion.Description, and is useful for accessing the field via an interface.
+func (v *CreateQuestionCreateQuestion) GetDescription() string { return v.Description }
+
+// GetPollingInterval returns CreateQuestionCreateQuestion.PollingInterval, and is useful for accessing the field via an interface.
+func (v *CreateQuestionCreateQuestion) GetPollingInterval() SchedulerPollingInterval {
+	return v.PollingInterval
+}
+
+// GetQueries returns CreateQuestionCreateQuestion.Queries, and is useful for accessing the field via an interface.
+func (v *CreateQuestionCreateQuestion) GetQueries() []CreateQuestionCreateQuestionQueriesQuestionQuery {
+	return v.Queries
+}
+
+// GetTags returns CreateQuestionCreateQuestion.Tags, and is useful for accessing the field via an interface.
+func (v *CreateQuestionCreateQuestion) GetTags() []string { return v.Tags }
+
+// GetCompliance returns CreateQuestionCreateQuestion.Compliance, and is useful for accessing the field via an interface.
+func (v *CreateQuestionCreateQuestion) GetCompliance() []CreateQuestionCreateQuestionComplianceQuestionComplianceMetaData {
+	return v.Compliance
+}
+
+// CreateQuestionCreateQuestionComplianceQuestionComplianceMetaData includes the requested fields of the GraphQL type QuestionComplianceMetaData.
+type CreateQuestionCreateQuestionComplianceQuestionComplianceMetaData struct {
```

Comment:
> Is there anyway to control the naming scheme or have it reuse parts that are the same? This type is duplicated elsewhere, which may not be the biggest deal - it's generated code. Where we could re-use types or instruct it to do so it would be nice.

The naming is a combination of different types in the schema and the `genql` chosen way to reduce ambiguity in output, where each individual GraphQL query may return different fields for the same type and leave some fields with whatever go's default values are.

Once I get to adding more of the Compliance Framework related entities, I'll keep on a eye out for this, but for this PR: following the GraphQL pattern of not requesting data that isn't used for efficiency and the terraform pattern of not fetching and setting anything that isn't supposed to be different from what was sent, I should have just removed them.